### DISCUSSION TITLE
Using digital elevation models in position estimation

### DISCUSSION THREAD

**Discussion by giaminhhoang** [JonasBchrt/snapshot-gnss-algorithms#5] [Q&A] [answered]
Hello,
In GNSS position estimation, known altitude is useful to improve the accuracy or constraint the results, and I found that your algorithm is using it. I am wondering how can you compile the `*.hgt` files in` /digital_elevation_model` to a full `egm96-5.pgm` 2-D map. I guess to have the full map, the number of *.hgt files should be more.
Could you share some insights?
Thanks a lot.
Best.

**Comment by JonasBchrt** [comment, marked as answer]
That's correct.
You can download more `.hgt` files from here: https://e4ftl01.cr.usgs.gov/MEASURES/SRTMGL1.003/2000.02.11/
The ones in the repository cover just Oxford.

Alternatively, use ETOPO1 for global coverage.

(More details in the doc string of `eph_util.get_elevation(...)`: https://github.com/JonasBchrt/snapshot-gnss-algorithms/blob/422d679237b6aa0b891943a0b1bab59fc85e5dfe/eph_util.py#L3545 )

Does this help?

**Reply by giaminhhoang** [reply]
Thanks Jonas. It helps a lot.

**Comment by giaminhhoang** [comment]
Though the global coverage of ETOPO1 is quite interesting, its 1-arc minute resolution is not enough for my study. So I have to come back to SRTM 1 Arc-Second. The link you provided to download .hgt files for this model while https://earthexplorer.usgs.gov/ gives only dted, bil or GeoTIFF formats. But I understood that they come from the same SRTM.
Do you know which one I should use? Thanks in advance.

**Reply by JonasBchrt** [reply]
I used the `.hgt` files because they are supported by [the `python-srtm` package](https://pypi.org/project/python-srtm/).
I believe DTED and GeoTIFF should be able to store the same information. (Don't know about BIL.) There are Python packages for those formats, too. (If Python is your language of choice.) They might differ in file size, though. I believe I have also seen converters for those file types out there.

**Reply by giaminhhoang** [reply]
Thanks a lot for your quick reply. I am using Matlab and just founding this project using Matlab to read the `.hgt` file https://fr.mathworks.com/matlabcentral/fileexchange/36379-readhgt-import-download-nasa-srtm-data-files-hgt.
